addons/mail/static/tests/suggestion/suggestion_tests.js

```diff
@@ -319,3 +319,17 @@ QUnit.test("Current user that is a follower should be considered as such", async
         before: [".o-mail-Composer-suggestion", { text: "Person A(a@test.com)" }],
     });
 });
+
+QUnit.test("Suggestions that begin with the search term should have priority", async () => {
+    const pyEnv = await startServer();
+    pyEnv["res.partner"].create([{ name: "Party Partner" }, { name: "Best Partner" }]);
+    const { openFormView } = await start();
+    openFormView("res.partner", pyEnv.currentPartnerId);
+    await click("button", { text: "Send message" });
+    await insertText(".o-mail-Composer-input", "@part");
+    await contains(".o-mail-Composer-suggestion", { count: 2 });
+    await contains(".o-mail-Composer-suggestion", {
+        text: "Party Partner",
+        before: [".o-mail-Composer-suggestion", { text: "Best Partner" }],
```

Comment:
I would suggest adding a section to test if by default `Best Partner` is before `Party Partner`, which should be the case. So there should be an extra part before as: `await insertText(".o-mail-Composer-input", "@");` and `Best Partner` before `Party Partner`.